sharktuner/boo_tuner.py

```diff
@@ -0,0 +1,124 @@
+"""
+Generate BOO dispatches for the specified kernels and tune them. Kernel
+configurations follow the MIOpen driver format.
+
+Example usage:
+python boo_tuner.py \
+  convbfp16 -n 6 -c 112 -H 1 -W 1 -k 448 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -F 1 \
+ --output-td-spec tuning_spec.mlir --num-candidates 5000 --devices='hip://0'
+"""
+
+import argparse
+import shlex
+from pathlib import Path
+import os
+import tempfile
+import subprocess
+import shutil
+import traceback
+
+from iree.turbine.kernel.boo import runtime as boo_runtime
+from iree.turbine.kernel.boo.driver.launch import get_launchable
+from iree.turbine.kernel.boo.op_exports.conv import ConvParser as mio
+
+from model_tuner.model_tuner import main as tuner_main
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--commands-file", type=str, help="read commands from file")
+    parser.add_argument("--output-td-spec", type=Path, default="tuning-spec.mlir")
+    parser.add_argument("--starter-td-spec", type=Path)
+    parser.add_argument("--num-candidates", type=int, default=100)
+    parser.add_argument("--devices", type=str, default="hip://0")
+    parser.add_argument(
+        "--tmp-dir", type=str, default="", help="directory to save temporary files"
+    )
+    args, extra_cli_args = parser.parse_known_args()
+
+    if args.commands_file:
+        with open(args.commands_file) as f:
+            mio_file_args = [
+                shlex.split(s) for s in f.readlines() if not s.startswith("#")
+            ]
+    else:
+        mio_file_args = [[]]  # use CLI arguments
```

Comment:
What does `mio` stand for?

```suggestion
    mio_file_args = [[]]  # Use CLI arguments.
    if args.commands_file:
        with open(args.commands_file) as f:
            mio_file_args = [
                shlex.split(s) for s in f.readlines() if not s.startswith("#")
            ]
```